cli/src/merge_tools/mod.rs

```diff
@@ -264,6 +264,21 @@ pub fn run_mergetool(
     let content = file_merge.extract_as_single_hunk(tree.store(), repo_path);
 
     let editor = get_merge_tool_from_settings(ui, settings)?;
+    match editor {
+        MergeTool::Internal => unimplemented!("run_mergetool with internal mergetool"),
+        MergeTool::External(editor) => {
+            run_mergetool_external(editor, content, repo_path, conflict, tree)
+        }
+    }
+}
+
+fn run_mergetool_external(
+    editor: ExternalMergeTool,
+    content: jj_lib::merge::Merge<jj_lib::files::ContentHunk>,
+    repo_path: &RepoPath,
+    conflict: jj_lib::merge::Merge<Option<TreeValue>>,
+    tree: &Tree,
+) -> Result<TreeId, ConflictResolveError> {
```

Comment:
nit: passing `&ExternalMergeTool` by reference would make more sense since it's a configuration data.